text/0000-rustc-bug-fix-procedure.md

```diff
@@ -0,0 +1,273 @@
+- Feature Name: N/A
+- Start Date: 2016-04-22
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Defines a best practices procedure for making bug fixes or soundness
+corrections in the compiler that can cause existing code to stop
+compiling.
+
+# Motivation
+[motivation]: #motivation
+
+From time to time, we encounter the need to make a bug fix, soundness
+correction, or other change in the compiler which will cause existing
+code to stop compiling. When this happens, it is important that we
+handle the change in a way that gives users of Rust a smooth
+transition. What we want to avoid is that existing programs suddenly
+stop compiling with opaque error messages: we would prefer to have a
+gradual period of warnings, with clear guidance as to what the problem
+is, how to fix it, and why the change was made. This RFC describes the
+procedure that we have been developing for handling breaking changes
+that aims to achieve that kind of smooth transition.
+
+One of the key points of this policy is that (a) warnings should be
+issued initially rather than hard errors if at all possible and (b)
+every change that causes existing code to stop compiling will have an
+associated tracking issue. This issue provides a point to collect
+feedback on the results of that change. Sometimes changes have
+unexpectedly large consequences or there may be a way to avoid the
+change that was not considered.  In those cases, we may decide to
+change course and roll back the change, or find another solution (if
+warnings are being used, this is particularly easy to do).
+
+### What qualifies as a bug fix? 
+
+Note that this RFC does not try to define when a breaking change is
+permitted. That is already covered under [RFC 1122][]. This document
+assumes that the change being made is in accordance with those
+policies. Here is a summary of the conditions from RFC 1122:
+
+- **Soundness changes:** Fixes to holes uncovered in the type system.
+- **Compiler bugs:** Places where the compiler is not implementing the
+  specified semantics found in an RFC or lang-team decision.
```

Comment:
We do not have an RFC for the whole language as it was before RFC process was introduced. Perhaps this should also include things like the [rust reference](http://doc.rust-lang.org/reference.html)?